pkg/cmd/cli/cmd/newapp.go

```diff
@@ -271,15 +274,56 @@ func setLabels(labels map[string]string, result *newcmd.AppResult) error {
 	return nil
 }
 
+// isInvalidTriggerError returns true if the given error is
+// a validation error that contains 'trigger' in its error message.
+// This error is returned from an older server when
+// it doesn't recognize a newer trigger type.
+func isInvalidTriggerError(err error) bool {
+	if !kapierrors.IsInvalid(err) {
+		return false
+	}
+	statusErr, ok := err.(*kapierrors.StatusError)
+	if !ok {
+		return false
+	}
+	return strings.Contains(statusErr.Status().Message, "trigger")
+}
+
+// retryBuildConfig determines if the given error is caused by an invalid trigger
+// error on a BuildConfig. If that is the case, it will remove all triggers with a
+// type that is not in the whitelist for an older server.
+func retryBuildConfig(info *resource.Info, err error) runtime.Object {
+	triggerTypeWhiteList := map[buildapi.BuildTriggerType]struct{}{
+		buildapi.GitHubWebHookBuildTriggerType:  struct{}{},
+		buildapi.GenericWebHookBuildTriggerType: struct{}{},
+		buildapi.ImageChangeBuildTriggerType:    struct{}{},
```

Comment:
i guess i'm not actually sure what happens as far as serialization but all @rhcarvalho changed was the validation logic to tolerate the unknown Type.  So i assume we'd end up with a trigger entry in the array with a Type="Foo" and then trigger.Foo would be totally empty (once "Foo" was a valid trigger type that we added a field for).

so i guess what we should be doing is dropping the entire Trigger entry in the array.  @rhcarvalho